server/src/ingestion.rs

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2022 The MiniModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Support for data ingestion from an MQTT broker.
+//!
+//! To use the ingestor, run the "ingestor.start()" method. This method first creates the client. Then
+//! it uses the created client to connect to the broker and subscribe to the specified topics. The
+//! connection message stream is then looped over to ingest the messages that are published to the topics.
+
+use std::{process, time::Duration};
+
+use futures::{executor::block_on, stream::StreamExt};
+use paho_mqtt as mqtt;
+use paho_mqtt::{AsyncClient, AsyncReceiver, Message};
+
+use crate::storage::StorageEngine;
+
+/// A single MQTT client that can subscribe to `broker` and ingest messages from `topics`. Note that
+/// after creation, the ingestor needs to be started to ingest messages.
+pub struct Ingestor {
```

Comment:
Unclear why all of the struct's field and method parameters have the lifetime [`'static`](https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html) instead of defining a [lifetime for the `Ingestor` struct.](https://doc.rust-lang.org/rust-by-example/scope/lifetime/struct.html)

Reply:
Simplified the lifetime of the struct fields by changing to types to `String` and `Vec`.